java.source.base/src/org/netbeans/modules/java/source/save/CasualDiff.java

```diff
@@ -5119,6 +5128,7 @@ private int diffTreeImpl0(JCTree oldT, JCTree newT, JCTree parent /*used only fo
 
         if (oldT == newT)
             return elementBounds[0];
+        assert oldT != null;
```

Comment:
Assignment: If oldT is null here, what would the value of "newT" be?

Reply:
I am not sure I understand your comment. `oldT` cannot be `null` at line 5131. Assume that diffTreeImpl0() is called with `oldT = null`. To get past the if statement on line 5126, `newT` would have to be `null`. But then, `oldT == newT` (`null == null`), so the method would return `elementBounds[0]` from the preceding if.

Reply:
Exactly, so the assertion will always fail and is thus redundant. Please remove it.

Reply:
The assertion is there to help out NetBeans' null checking analysis (and also humans, as the logic is somewhat tricky), so that NetBeans does not show a "Dereferencing possible null pointer" warning on `oldT.getTag()` on line 5176.

Reply:
Ah! So the code is "somewhat tricky" right ;-) That's an issue. Let's try to make it more readable. What about something like:

```
if (oldT == null) {
    if(newT == null) {
       // ...
    } else {
      // ...
    }
}
```
Would that enhance readability?

Reply:
I changed it to:

```java
if (oldT == newT) {
    return elementBounds[0];
}

if (oldT == null) {
    throw new IllegalArgumentException("Null is not allowed in parameters.");
}
```

To start with `if (oldT == null)` would require:

```java
if (oldT == null) {
    if (newT == null) {
        return elementBounds[0];
    } else {
        throw new IllegalArgumentException("Null is not allowed in parameters.");
    }
} else if (oldT == newT) {
    return elementBounds[0];
}
```

.. but this duplicates the `return elementBounds[0];` line.